### PR TITLE
ParallelRunner catches exceptions correctly and has failure policies.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@sahilTakiar can you review?

**Comment by sahilTakiar** [comment]
Can you rebase your commit, I believe https://github.com/linkedin/gobblin/pull/661 may be trying to do something similar

**Comment by kadaan** [comment]
@ibuenros I just changed this class to deal with the issue i reported last night.

**Comment by ibuenros** [comment]
@kadaan @sahilTakiar Thanks for pointing that out. I changed the PR to include those changes. This PR is adding logging for those exceptions and still including fail policies.

File: gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java [hunk 1]
```diff
@@ -78,14 +80,35 @@
...
+  }
+
+  public ParallelRunner(int threads, FileSystem fs, FailPolicy failPolicy) {
     this.executor = Executors.newFixedThreadPool(threads,
         ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("ParallelRunner")));
     this.fs = fs;
+    this.failPolicy = failPolicy;
+  }
+
+  public static enum FailPolicy {
```
Comment (kadaan) [inline gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java]: Can you add javadocs around the intent of these.  For ParallelRunner I think it is really important to be crystal clear what the expectations are.

**Comment by ibuenros** [comment]
@kadaan added documentation on the policies.

**Comment by sahilTakiar** [comment]
LGTM

**Comment by kadaan** [comment]
LGTM

**Pull request merged**